packages/nx/src/daemon/cache.ts

```diff
@@ -43,6 +43,23 @@ export async function safelyCleanUpExistingProcess(): Promise<void> {
   if (daemonProcessJson && daemonProcessJson.processId) {
     try {
       process.kill(daemonProcessJson.processId);
+      await new Promise<void>((resolve, reject) => {
+        let count = 0;
+        const interval = setInterval(() => {
+          try {
+            process.kill(daemonProcessJson.processId, 0);
+          } catch (e) {
+            clearInterval(interval);
+            resolve();
+          }
+          if ((count += 1) > 200) {
+            clearInterval(interval);
+            reject(
+              `Daemon process ${daemonProcessJson.processId} didn't exit after 2 seconds.`
+            );
+          }
+        }, 10);
+      });
```

Comment:
Can you leave a comment here?

The code looks a little weird at first.

On line 50 we're actually checking if the process still exists right? And it'll error if the process does not which is actually a success for what we're doing here?